### PR TITLE
feat: add relations fe state management conceptual documentation

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Adds conceptual documentation of how relations FE state management works

### Why is it needed?

* I might get hit by a bus and sometimes code can be overwhelming.

### Related issue(s)/PR(s)

* relates to CONTENT-661

### CODE REVIEW

**Comment by joshuaellis** [comment]
Would you rather contribute to this PR and then merge? Or incrementally update these docs @gu-stav & @ronronscelestes?

File: docs/docs/core/content-manager/relations.mdx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+Inside the reducer we reduce the array of `relationalFieldPaths` to an object with the `initialValues` clone as
+as the base. If there is `modifiedData` in the browser i.e. you've made changes to the entity and saved those changes,
+we just replace the first level of the field with the `modifiedData` so the data structure is preserved and we're not
+loosing the relations we had already loaded in the component. If the first part of the path is highlighted as the
+`relationalField` then we simply replace that intial object with an empty array.
+
+However, if the first part of the path is either a repeatable component, a dynamic zone or a regular component then we
+recursively find the relation fields and replace the object with an array. This is handled by the `findLeafByPathAndReplace`
+utility function. This function in short, takes and end path (in this case the relational field) and a primitive to replace
```
Comment (ronronscelestes) [inline docs/docs/core/content-manager/relations.mdx:104]: > This function in short, takes an(**d**) end path
👀

**Review by ronronscelestes — COMMENTED** [review]
yay documentation ✨
found a small typo, other than that LGTM 🙏

> Would you rather contribute to this PR and then merge? Or incrementally update these docs

no strong opinion there, as you both prefer

**Review by ronronscelestes — APPROVED** [review]
LGTM, thanks!

**Comment by joshuaellis** [comment]
@alexandrebodin or @Convly do you know why these tests aren't running? I assume it's because all i've done it update documentation, but they're required so I can't merge the PR 🥲

**Comment by alexandrebodin** [comment]
Yeah that's something we can't manage with github actions that's a bit annoying. if you are ready I'll merge it for now but we have to figure sth out

**Comment by joshuaellis** [comment]
yes please! 

Could we use [paths](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore)? There's a github action if we need a bit more "power" – https://github.com/dorny/paths-filter

**Comment by alexandrebodin** [comment]
> yes please!
> 
> Could we use [paths](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore)? There's a github action if we need a bit more "power" – https://github.com/dorny/paths-filter

The problem isn't about paths but required checks.  They are define in the repo settings and set globally for PRs regarless of the jobs that run. we already have path filters to execute them but that's becaue they don't execute that github isn't happy because they are always required for all PRs. The only solution AFAIK is to make the checks non required which isn't great...

**Comment by alexandrebodin** [comment]
> Whilst i think this is a lousy solution, i'll leave this doc here on what github suggest – [skip checks](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks)
> 
> which, basically suggests making duplicate actions so github "thinks" they've passed 😓

Yeah... if I can make this in one matrix why not.... I'll think about it. thanks for sharing

**Pull request merged**